plutus-chain-index/app/Marconi.hs

```diff
@@ -174,50 +156,106 @@ getUtxoUpdate slot txs =
                  , _slotNo  = slot
                  }
 
-utxoIndexer
-  :: FilePath
+data Coordinator = Coordinator
+  { _channel      :: TChan (ChainSyncEvent (BlockInMode CardanoMode))
+  , _barrier      :: QSemN
+  , _indexerCount :: Int
+  }
+
+initialCoordinator :: Int -> IO Coordinator
+initialCoordinator indexerCount =
+  Coordinator <$> newBroadcastTChanIO
+              <*> newQSemN 0
+              <*> pure indexerCount
+
+datumWorker
+  :: Coordinator
+  -> TChan (ChainSyncEvent (BlockInMode CardanoMode))
+  -> FilePath
+  -> IO ()
+datumWorker Coordinator{_barrier} ch path = Datum.open path (Datum.Depth 2160) >>= innerLoop
```

Comment:
The security param `2160` shouldn't be hardcoded right?

Reply:
I am not sure why it shouldn't be hardcoded in this case. Would you like to expose it as a command-line argument? One for all indexers, or one command-line argument for each? I can write the code if you think it's important to do so.